daemon/cmd/endpoint.go

```diff
@@ -413,6 +414,21 @@ func (d *Daemon) createEndpoint(ctx context.Context, owner regeneration.Owner, e
 		}
 	}
 
+	if ep.K8sNamespaceAndPodNameIsSet() && d.clientset.IsEnabled() {
+		_, _, _, _, annotations, err := d.fetchK8sMetadataForEndpoint(ep.K8sNamespace, ep.K8sPodName)
+		if err != nil {
+			ep.Logger("api").WithError(err).Warning("Unable to fetch kubernetes labels")
+		}
+		if hwAddr, ok := annotations["cni.cilium.io/hwAddr"]; ok {
+			m, err := mac.ParseMAC(hwAddr)
+			if err != nil {
+				log.WithField(logfields.K8sPodName, epTemplate.K8sNamespace+"/"+epTemplate.K8sPodName).
+					WithError(err).Error("Unable to parse MAC address")
+				return invalidDataError(ep, err)
+			}
+			ep.SetMac(m)
+		}
+	}
```

Comment:
I think that these changes should be combined with the `fetchK8sMetadataForEndpoint` code below. No need to do that step twice.